cli/src/main/java/org/jboss/as/cli/impl/ReadlineConsole.java

```diff
@@ -88,6 +98,225 @@
+                        builder.colorize(position, Color.DEFAULT, Color.DEFAULT, false);
+                        builder.build().run();
+                    }
+                    openPositions.clear();
+                }
+            }
+
+        }
+
+        void clear(int position) {
+            if (lastLine != null) {
+                Line.CursorTransactionBuilder builder = lastLine.newCursorTransactionBuilder();
+                builder.colorize(position, Color.DEFAULT, Color.DEFAULT, false);
+                builder.build().run();
+            }
+        }
+
+        void match(Line line) {
+            lastLine = line;
+            clear();
+
+            char separator = (char) line.getCharacterAtCursor();
+
+            if (endSeparators.contains(separator)) {
+                String l = line.getLineToCursor();
+                char startChar = endStartSeparators.get(separator);
+                int index = findStart(l, startChar, separator);
+                if (index == -1) {
+                    return;
+                }
+
+                scan(line, index, line.getCurrentCharacterIndex());
+                lastIndex = colorize(line, index);
+            } else if (startSeparators.contains(separator)) {
+                String l = line.getLineFromCursor();
+                char endChar = startEndSeparators.get(separator);
+                int index = findEnd(l, line.toString().length(), separator, endChar);
+                if (index == -1) {
+                    return;
+                }
+
+                scan(line, line.getCurrentCharacterIndex(), index);
+                lastIndex = colorize(line, index);
+            } else {
+                lastIndex = -1;
+            }
+        }
+
+        void scan(Line line, int start, int end) {
+            List<AbstractMap.SimpleEntry<Character,Integer>> brackets = new ArrayList<>();
+
+            // Collect the position of all brackets in the line
+            for (int i = start; i <= end; i++) {
+                char currentChar = line.getCharacterAtPosition(i);
+                if (endSeparators.contains(currentChar) || startSeparators.contains(currentChar)) {
+                    brackets.add(new AbstractMap.SimpleEntry<>(currentChar, i));
+                }
+            }
+
+            // If there's only one bracket in the line, mark it and return
+            if (brackets.size() == 1) {
+                AbstractMap.SimpleEntry<Character, Integer> solo = brackets.get(0);
+                if (endStartSeparators.containsKey(solo.getKey()) || startEndSeparators.containsKey(solo.getKey())) {
+                    colorizeOpenPair(line, solo.getValue());
+                    return;
+                }
+            }
+
+            Set<AbstractMap.SimpleEntry<Character, Integer>> closedBrackets = new HashSet<>();
+            Set<AbstractMap.SimpleEntry<Character, Integer>> openBrackets = new HashSet<>();
```

Comment:
@iweiss Would you mind using a collection that preserves some reasonable ordering (e.g. LinkedHashSet) for closedBrackets and openBrackets? I am writing tests and only way to test this is to move the cursor through the command and then check ansi sequence that describes cursor movement and color changes. However, this sequence is dependent on HashSet which ordering is not guaranteed. I believe this change wouldn't affect functionality, but would simplify testing and tests maintenance.

Reply:
Not a problem @tterem. PR updated.